core/dnsserver/onstartup_test.go

```diff
@@ -0,0 +1,9 @@
+package dnsserver
+
+import "testing"
+
+func TestCheckDomainName(t *testing.T) {
+	if !checkDomainName("dns://example.com:53") {
+		t.Fatalf(`Something went wrong, Now domain name doesn't match with Regular-Expression`)
+	}
```

Comment:
There should several positive and negative test cases.  i.e. some cases that you expect to get a true result, and some that expect a false result.  

The test failure message should describe the test case and the expected result.  
e.g. `Expected true for "an-example.com"`
or `Expected false for "example-?&^%$.com"`